foundation-ui/services/Hooks.js

```diff
@@ -0,0 +1,153 @@
+function addListener(store, hook, listener, priority = 10) {
+  if (typeof priority !== 'number') {
+    priority = 10;
+  }
+
+  if (store[hook] === undefined) {
+    store[hook] = {};
+  }
+
+  if (store[hook][priority] === undefined) {
+    store[hook][priority] = [];
+  }
+
+  store[hook][priority].push(listener);
+}
+
+function removeListener(store, hook, listener, priority = 10) {
```

Comment:
In addition, the Hooks now doesn't rely on the EventEmitter, since we can now remove actions and filters